src/CattyTests/DataModel/VariablesContainerTest.swift

```diff
@@ -96,4 +96,145 @@ final class VariablesContainerTest: XCTestCase {
+        allVariable = container.allVariables() as! [UserVariable]
+
+        XCTAssertEqual(2, allVariable.count)
+        XCTAssertTrue(allVariable[0].name == userVariable1.name)
+        XCTAssertTrue(allVariable[1].name == userVariable2.name)
+    }
+
+    func testAllList() {
+        let objectA = SpriteObject()
+        objectA.name = "testObjectA"
+
+        let list1 = UserVariable()
+        list1.name = "testName1"
+        list1.isList = true
+
+        let list2 = UserVariable()
+        list2.name = "testName2"
+        list2.isList = true
+
+        let container = VariablesContainer()
+
+        container.addObjectList(list1, for: objectA)
+        var allList = container.allLists() as! [UserVariable]
+
+        XCTAssertEqual(1, allList.count)
+        XCTAssertTrue(allList[0].name == list1.name)
+
+        container.addObjectList(list2, for: objectA)
+        allList = container.allLists() as! [UserVariable]
+
+        XCTAssertEqual(2, allList.count)
+        XCTAssertTrue(allList[0].name == list1.name)
+        XCTAssertTrue(allList[1].name == list2.name)
+    }
+
+    func testAllVariableAndAllList() {
+        let objectA = SpriteObject()
+        objectA.name = "testObjectA"
+
+        let userVariable = UserVariable()
+        userVariable.name = "testVariable"
+
+        let list = UserVariable()
+        list.name = "testList"
+        list.isList = true
+
+        let container = VariablesContainer()
+
+        container.addObjectVariable(userVariable, for: objectA)
+        var allListAndALlVariable = container.allVariablesAndLists() as! [UserVariable]
+
+        XCTAssertEqual(1, allListAndALlVariable.count)
+        XCTAssertTrue(allListAndALlVariable[0].name == userVariable.name)
+        XCTAssertFalse(allListAndALlVariable[0].isList)
+
+        container.addObjectList(list, for: objectA)
+        allListAndALlVariable = container.allVariablesAndLists() as! [UserVariable]
+
+        XCTAssertEqual(2, allListAndALlVariable.count)
+        XCTAssertTrue(allListAndALlVariable[0].name == userVariable.name)
+        XCTAssertTrue(allListAndALlVariable[1].name == list.name)
+        XCTAssertFalse(allListAndALlVariable[0].isList)
+        XCTAssertTrue(allListAndALlVariable[1].isList)
+    }
+
+    func testObjectVariablesForObject() {
+        let objectA = SpriteObject()
+        objectA.name = "testObjectA"
+
+        let userVariable1 = UserVariable()
+        userVariable1.name = "testVariable1"
+
+        let userVariable2 = UserVariable()
+        userVariable2.name = "testVariable2"
+
+        let container = VariablesContainer()
+
+        container.addObjectVariable(userVariable1, for: objectA)
+        var variables = container.objectVariables(for: objectA) as! [UserVariable]
+
+        XCTAssertEqual(1, variables.count)
+        XCTAssertTrue(variables[0].name == userVariable1.name)
+
+        container.addObjectVariable(userVariable2, for: objectA)
```

Comment:
Please add another object as well to document that the variables of this object are not part of the result.